src/pexprs.js

```diff
@@ -33,6 +35,14 @@ PExpr.prototype.withInterval = function(interval) {
   return this;
 };
 
+// Record the start and end index of the source for this expression, relative to an implicit
+// source string. Similar to `withInterval`, but can be used when it's not possible to pass
+// an Interval instance (e.g., inside a recipe).
+PExpr.prototype.fromSourceAt = function(startIdx, endIdx) {
```

Comment:
Ah, good catch. This was left over from my initial implementation, which I wasn't happy with. Fixed.